applications/openshift/api-server/api_server_admission_control_plugin_AlwaysPullImages/oval/shared.xml

```diff
@@ -0,0 +1,29 @@
+<def-group>
+  <definition class="compliance" id="api_server_admission_control_plugin_AlwaysPullImages" version="1">
+    <metadata>
+      <title>Enable the AlwaysPullImages Admission Control Plugin</title>
+      <affected family="unix">
+        <platform>multi_platform_ocp</platform>
+      </affected>
+      <description>Ensure credentials are required to pull images.</description>
+    </metadata>
+    <criteria operator="AND">
+      <criterion comment="AlwaysPullImages is configured" test_ref="test_api_server_admission_control_plugin_AlwaysPullImages" />
+    </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test check="all" check_existence="all_exist" comment="AlwaysPullImages is configured" id="test_api_server_admission_control_plugin_AlwaysPullImages" version="1">
+    <ind:object object_ref="object_api_server_admission_control_plugin_AlwaysPullImages" />
+    <ind:state state_ref="state_api_server_admission_control_plugin_AlwaysPullImages" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object id="object_api_server_admission_control_plugin_AlwaysPullImages" version="1">
+    <ind:filepath>/etc/origin/master/master-config.yaml</ind:filepath>
+    <ind:pattern operation="pattern match">^[\s]*admissionConfig\:(?:[^\n]*\n+)+?[\s]*pluginConfig\:(?:[^\n]*\n+)+?[\s]*AlwaysPullImages\:[\s]*[\n]+[\s]*configuration\:[\s]*[\n]+[\s]*kind\:[\s]+DefaultAdmissionConfig[\s]*[\n]+[\s]*apiVersion\:[\s]+v1[\s]*[\n]+[\s]*disable\:[\s]+(\S+)[\s]*$</ind:pattern>
```

Comment:
Current regex evaluates more than ``disable: false``, matching on things like ``apiVersion`` and ``DefaultAdmissionConfig``. Are those relevant to the configuration check?

Reply:
I'd say so for a valid configuration check.

Reply:
should they be independent configuration checks then?

Reply:
in the same way PAM options are broken out in RHEL

Reply:
@shawndwells preferably not. These checks are already complicated as is. Let's not add way more complication to it.

Reply:
the xccdf is worded vaguely enough to scope the subconfiguration elements. Might be something to review in future editions of the OCP content.